### PR TITLE
[XLA] Use non-inlined functions where inlining causes problems

### PR DESCRIPTION
**Pull request by DavidNorman**
On my OS/X machine these two functions, currently defined in the header, no not get marked `weak`, and cause a compilation error.

They do not appear to need to be defines in a header file.

This change removes them from the header and places them into the `cc` file.

### CODE REVIEW

**Comment by tensorflowbutler** [comment]
Nagging Assignee @rmlarsen: It has been 17 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by DavidNorman** [comment]
@rmlarsen Hi - could I get this reviewed.  I think that it is a fairly non-controversial change.

**Comment by DavidNorman** [comment]
@tatatodd Hi there.  Could we get this reviewed and merged?  I think it is a fairly non-controversial change.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @rmlarsen: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by DavidNorman** [comment]
@tatatodd @rmlarsen Hello.   This is over 2 months old now.  Can someone look at it?

**Comment by DavidNorman** [comment]
@vrv @hawkinsp Hi guys.  I'm guessing that the chaps who are down to review this don't know they are or get notifications.  Is there a way of getting someone else to review it?  It is over 2 months old now.  It's not a complicated thing, and it doesn't fail on linux, only OS/X.

**Review by vrv — APPROVED** [review]

**Comment by vrv** [comment]
Sorry about the delay.  My guess is with 262 active PRs, many of those here listed also on them, these are getting lost in the noise.  This is a simple change so LGTM, even though I'm pretty unfamiliar with the code myself.  Thanks for the contribution!

**Comment by DavidNorman** [comment]
Hi @vrv, I don't think that these failures are to do with the change.  The XLA ones look like a resource issue (something to do with remote cache), and the ubuntu sanity fails from time to time anyway.

could we try to run the unit tests again?

**Comment by vrv** [comment]
Yes, let me try again, thanks.

**Comment by DavidNorman** [comment]
Ok. So there is a compilation error, which is odd as it doesn't appear on either of our build environments.  Could be something to do with compiling with the GPU enabled.

```
tensorflow/compiler/xla/tests/client_library_test_base.cc:579:1: error: 'std::unique_ptr<xla::GlobalData> xla::ClientLibraryTestBase::CreateParameterAndTransferLiteral(tensorflow::int64, const xla::Literal&, const string&, const xla::DeviceHandle*, xla::XlaBuilder*, xla::XlaOp*)' previously defined here
 ClientLibraryTestBase::CreateParameterAndTransferLiteral(
```

**Comment by DavidNorman** [comment]
ah ha - it turns out that this problem has been fixed by someone else already.

**Comment by vrv** [comment]
:(.  Thanks for checking and dealing with so many delays, your effort is appreciated (by me at least!).
